gentoo_update/shell_runner.py

```diff
@@ -0,0 +1,139 @@
+import os
+import sys
+import shlex
+import logging
+import subprocess
+from datetime import datetime
+
+
+class ShellRunner:
+    def __init__(self, quiet):
+        self.quiet = True if quiet == "y" else False
+
+        self.timestamp = datetime.now().strftime("%Y-%m-%d-%H-%M")
+        self.log_dir = "/var/log/gentoo_update"
+        self.log_filename = f"{self.log_dir}/log_{self.timestamp}"
+        self.logger = self.initiate_logger()
+
+    def initiate_logger(self):
+        """
+        Create a logger with two handlers:
+            1. terminal output
+            2. file output
+        Both handlers have the same logging level (INFO)
+        and share the same formatter.
+        Formatters include timestamp, log level and the message.
+
+        Returns:
+            logging.Logger: Configured logger.
+            log_filename: Log filename.
+        """
+        if not os.path.exists(self.log_dir):
+            os.mkdir(self.log_dir)
+
+        logger = logging.getLogger(__name__)
+        logger.setLevel(logging.INFO)
+
+        formater = logging.Formatter(
+            "[%(asctime)s %(levelname)s] ::: %(message)s",
+            datefmt="%d-%b-%y %H:%M:%S",
+        )
+
+        if not self.quiet:
+            terminal_handler = logging.StreamHandler()
+            terminal_handler.setLevel(logging.INFO)
+            terminal_handler.setFormatter(formater)
+            logger.addHandler(terminal_handler)
+
+        file_handler = logging.FileHandler(self.log_filename)
+        file_handler.setLevel(logging.INFO)
+        file_handler.setFormatter(formater)
+        logger.addHandler(file_handler)
+
+        return logger
+
+    def _log_stream_output(self, stream_obj, type):
```

Comment:
Consider using static types in python for extra protections and readability

_log_stream_output(self, stream_obj:  logging.StreamHandler, type: str)